referencesrv/views.py

```diff
@@ -0,0 +1,191 @@
+# -*- coding: utf-8 -*-
+
+from flask import current_app, request, Blueprint, Response
+from flask_discoverer import advertise
+
+import json
+import urllib
+import re
+
+from referencesrv.parser.crf import CRFClassifierText, CRFClassifierXML
+from referencesrv.resolver.solve import solve_reference
+from referencesrv.resolver.hypotheses import Hypotheses
+
+
+bp = Blueprint('reference_service', __name__)
+
+
+def text_parser(reference):
+    """
+
+    :return:
+    """
+    if not hasattr(text_parser, "crf"):
```

Comment:
is this setting and attribute of itself? (function)

that seems funny

Reply:
yes for the function. and that is how it grantees there is only one time training.